src/hex_repo.erl

```diff
@@ -152,9 +154,15 @@ get_tarball(Name, Version) ->
 get_tarball(Name, Version, Options) ->
     Client = proplists:get_value(client, Options),
     Repo = proplists:get_value(repo, Options),
-    case get(Client, tarball_uri(Repo, Name, Version)) of
+    CacheDir = proplists:get_value(cache_dir, Options),
+
+    case get(Client, tarball_uri(Repo, Name, Version), make_headers(Options)) of
         {ok, {200, _Headers, Tarball}} ->
             {ok, Tarball};
```

Comment:
we also need a way to get an etag from the server, how about it returns `{ok, Tarball, [{etag, ETag}]}`? cc @ericmj

Reply:
OK, sounds good.